test/jdk/java/net/httpclient/CancelStreamedBodyTest.java

```diff
@@ -0,0 +1,432 @@
+        httpURI = "http://" + httpTestServer.serverAuthority() + "/http1/x/";
+
+        HttpsServer httpsServer = HttpsServer.create(sa, 0);
+        httpsServer.setHttpsConfigurator(new HttpsConfigurator(sslContext));
+        httpsTestServer = HttpTestServer.of(httpsServer);
+        httpsTestServer.addHandler(h1_chunkHandler, "/https1/x/");
+        httpsURI = "https://" + httpsTestServer.serverAuthority() + "/https1/x/";
+
+        // HTTP/2
+        HttpTestHandler h2_chunkedHandler = new HTTPSlowHandler();
+
+        http2TestServer = HttpTestServer.of(new Http2TestServer("localhost", false, 0));
+        http2TestServer.addHandler(h2_chunkedHandler, "/http2/x/");
+        http2URI = "http://" + http2TestServer.serverAuthority() + "/http2/x/";
+
+        https2TestServer = HttpTestServer.of(new Http2TestServer("localhost", true, sslContext));
+        https2TestServer.addHandler(h2_chunkedHandler, "/https2/x/");
+        https2URI = "https://" + https2TestServer.serverAuthority() + "/https2/x/";
+
+        serverCount.addAndGet(4);
+        httpTestServer.start();
+        httpsTestServer.start();
+        http2TestServer.start();
+        https2TestServer.start();
+    }
+
+    @AfterTest
+    public void teardown() throws Exception {
+        String sharedClientName =
+                sharedClient == null ? null : sharedClient.toString();
+        TRACKER.track(sharedClient);
+        sharedClient = null;
+        Thread.sleep(100);
+        AssertionError fail = TRACKER.check(500);
+        try {
+            httpTestServer.stop();
+            httpsTestServer.stop();
+            http2TestServer.stop();
+            https2TestServer.stop();
+        } finally {
+            if (fail != null) {
+                if (sharedClientName != null) {
+                    System.err.println("Shared client name is: " + sharedClientName);
+                }
+                throw fail;
+            }
+        }
+    }
+
+    /**
+     * A handler that slowly sends back a body to give time for the
+     * the request to get cancelled before the body is fully received.
+     */
+    static class HTTPSlowHandler implements HttpTestHandler {
+        @Override
+        public void handle(HttpTestExchange t) throws IOException {
+            try {
+                out.println("HTTPSlowHandler received request to " + t.getRequestURI());
+                System.err.println("HTTPSlowHandler received request to " + t.getRequestURI());
+
+                byte[] req;
+                try (InputStream is = t.getRequestBody()) {
+                    req = is.readAllBytes();
+                }
+                t.sendResponseHeaders(200, -1); // chunked/variable
+                try (OutputStream os = t.getResponseBody()) {
+                    // lets split the response in several chunks...
+                    String msg = (req != null && req.length != 0)
+                            ? new String(req, UTF_8)
+                            : BODY;
```

Comment:
As far as I can see, none of the test requests send a request body. So perhaps we can simplify this part to just always send this static `BODY` content without these additional checks?